pkg/kubernetes/errors.go

```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package kubernetes
+
+import (
+	"errors"
+	"io"
+	"net"
+	"syscall"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// IsRetryableError returns true if this Kubernetes API should be retried.
+func IsRetryableError(err error) bool {
+	if apierrors.IsTimeout(err) || apierrors.IsServerTimeout(err) || apierrors.IsInternalError(err) {
+		return true
+	}
+
+	if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
+		return true
+	}
+
+	netErr := &net.OpError{}
```

Comment:
Do we consistently use that form instead of `var netErr net.OpError`?

I wonder if we could add https://github.com/quasilyte/go-consistent to our pipeline to  enforce that

Reply:
these are different forms, `errors.As` requires pointer to pointer here

Reply:
Ok, `netErr := new(net.OpError)` then :)

Reply:
but the linter looks interesting :+1:

Reply:
I don't have strong preference towards one or another, not sure which one is most frequent